scripts/ssl_tts/ssl_tts_vc.py

```diff
@@ -0,0 +1,254 @@
+        assert args.target_audio_path is not None, "target_audio_path is required"
+        if args.out_path is None:
+            source_name = os.path.basename(args.source_audio_path).split(".")[0]
+            target_name = os.path.basename(args.target_audio_path).split(".")[0]
+            args.out_path = "swapped_{}_{}.wav".format(source_name, target_name)
+
+        source_target_out_pairs = [(args.source_audio_path, args.target_audio_path, args.out_path)]
+
+    out_paths = [r[2] for r in source_target_out_pairs]
+    out_dir = get_base_dir(out_paths)
+    if not os.path.exists(out_dir):
+        os.makedirs(out_dir)
+
+    ssl_model = ssl_tts.SSLDisentangler.load_from_checkpoint(args.ssl_model_ckpt_path, strict=False)
+    ssl_model = ssl_model.to(device)
+    ssl_model.eval()
+
+    vocoder = hifigan.HifiGanModel.load_from_checkpoint(args.hifi_ckpt_path).to(device)
+    vocoder.eval()
+
+    fastpitch_model = fastpitch_ssl.FastPitchModel_SSL.load_from_checkpoint(args.fastpitch_ckpt_path, strict=False)
+    fastpitch_model = fastpitch_model.to(device)
+    fastpitch_model.eval()
+    fastpitch_model.non_trainable_models = {'vocoder': vocoder}
+
+    wav_featurizer = WaveformFeaturizer(sample_rate=22050, int_values=False, augmentor=None)
+
+    use_unique_tokens = args.use_unique_tokens == 1
+    compute_pitch = args.compute_pitch == 1
+    compute_duration = args.compute_duration == 1
+
+    for source_target_out in source_target_out_pairs:
+        source_audio_path = source_target_out[0]
+        target_audio_paths = source_target_out[1].split(",")
+        out_path = source_target_out[2]
+
+        with torch.no_grad():
+            content_embedding1, _, duration1 = get_ssl_features_disentsngled(
+                ssl_model,
+                wav_featurizer,
+                source_audio_path,
+                emb_type="embedding_and_probs",
+                use_unique_tokens=use_unique_tokens,
+                device=device,
+            )
+
+            speaker_embedding2 = get_speaker_embedding(
+                ssl_model, wav_featurizer, target_audio_paths, duration=None, device=device
+            )
+
+            pitch_contour1 = None
+            if not compute_pitch:
+                pitch_contour1 = get_pitch_contour(load_wav(source_audio_path, wav_featurizer), compute_mean_std=True)[
+                    None
+                ]
+                pitch_contour1 = pitch_contour1.to(device)
+
+            wav_generated = fastpitch_model.generate_wav(
+                content_embedding1,
+                speaker_embedding2,
+                pitch_contour=pitch_contour1,
+                compute_pitch=compute_pitch,
+                compute_duration=compute_duration,
+                durs_gt=duration1,
+                dataset_id=0,
+            )
+            wav_generated = wav_generated[0][0]
+            soundfile.write(out_path, wav_generated, 22050)
```

Comment:
Should the sample rate here be a param based on the model config/user input?

Reply:
done.